lib/README.md

```diff
@@ -36,6 +36,70 @@ The [module system](https://nixos.org/manual/nixpkgs/#module-system) spans multi
 - [`options.nix`](options.nix): `lib.options` for anything relating to option definitions
 - [`types.nix`](types.nix): `lib.types` for module system types
 
+## PR Guidelines
+
+Follow these guidelines for proposing a change to the interface of `lib`.
+
+### Provide a Motivation
+
+Clearly describe why the change is necessary and its use cases.
+
+Make sure that the change benefits the user more than the added mental effort of looking it up and keeping track of its definition.
+If the same can reasonably be done with the existing interface,
+consider just updating the documentation with more examples and links.
+This is also known as the [Fairbairn Threshold](https://wiki.haskell.org/Fairbairn_threshold).
+
+Through this principle we avoid the human cost of duplicated functionality in an overly large library.
+
+### Make one PR for each change
+
+Don't have multiple changes in one PR, instead split it up into multiple ones.
+
+This keeps the conversation focused and has a higher chance of getting merged.
+
+### Name the interface appropriately
+
+When introducing new names to the interface, such as new function, or new function attributes,
+make sure to name it appropriately.
+
+Names should be self-explanatory and consistent with the rest of `lib`.
+If there's no obvious best name, include the alternatives you considered.
+If you don't know what name to choose, indicate this as a discussion point in the PR.
```

Comment:
"indicate as discussion point" sounds really heavy.

This sentence also conflicts a bit with the preceding one. Do we really think contributors can't come up with any name whatsoever? It actually seems condescending.

```suggestion
```

or

```suggestion
If you don't know what name to choose, feel free to say so in the PR description.
```

Reply:
I don't really mind either way, ping @h7x4

Reply:
@h7x4 I was hoping to get some thoughts :sweat_smile:

Reply:
Alright I'll just remove this sentence again. Let's merge before this stalls out for too long. Feel free to open follow-up PRs if there's something to improve.

Reply:
Sorry, the ping went by without me noticing. The thumbs up was intended for the entirety of roberth's comment, so if you removed the sentence I think all is good!

Reply:
Cool thanks!